File: ios/runanywhere-sdks-latest/sdk/runanywhere-swift/Sources/RunAnywhere/Data/Network/Models/Auth/AuthenticationResponse.swift
import Foundation

/// Response model for SDK authentication
/// Matches backend SDKAuthenticationResponse schema
public struct AuthenticationResponse: Codable, Sendable {
    public let accessToken: String
    public let refreshToken: String
    public let expiresIn: Int
    public let tokenType: String
    public let organizationId: String
    public let userId: String?
    public let deviceId: String?

    public init(
        accessToken: String,
        refreshToken: String,
        expiresIn: Int = 18000,
        tokenType: String = "Bearer",
        organizationId: String,
        userId: String? = nil,
        deviceId: String? = nil
    ) {
        self.accessToken = accessToken
        self.refreshToken = refreshToken
        self.expiresIn = expiresIn
        self.tokenType = tokenType
        self.organizationId = organizationId
        self.userId = userId
        self.deviceId = deviceId
    }

    enum CodingKeys: String, CodingKey {
        case accessToken = "access_token"
        case refreshToken = "refresh_token"
        case expiresIn = "expires_in"
        case tokenType = "token_type"
        case organizationId = "organization_id"
        case userId = "user_id"
        case deviceId = "device_id"
    }
}

/// Response model for token refresh (same as AuthenticationResponse)
public typealias RefreshTokenResponse = AuthenticationResponse

File: ios/runanywhere-sdks-latest/sdk/runanywhere-flutter/packages/runanywhere/lib/data/network/models/auth/authentication_response.dart
/// Response model for authentication.
///
/// Matches iOS `AuthenticationResponse` from RunAnywhere SDK.
class AuthenticationResponse {
  final String accessToken;
  final String deviceId;
  final int expiresIn;
  final String organizationId;
  final String refreshToken;
  final String tokenType;
  final String? userId;

  const AuthenticationResponse({
    required this.accessToken,
    required this.deviceId,
    required this.expiresIn,
    required this.organizationId,
    required this.refreshToken,
    required this.tokenType,
    this.userId,
  });

  factory AuthenticationResponse.fromJson(Map<String, dynamic> json) {
    return AuthenticationResponse(
      accessToken: json['access_token'] as String,
      deviceId: json['device_id'] as String,
      expiresIn: json['expires_in'] as int,
      organizationId: json['organization_id'] as String,
      refreshToken: json['refresh_token'] as String,
      tokenType: json['token_type'] as String,
      userId: json['user_id'] as String?,
    );
  }

  Map<String, dynamic> toJson() => {
        'access_token': accessToken,
        'device_id': deviceId,
        'expires_in': expiresIn,
        'organization_id': organizationId,
        'refresh_token': refreshToken,
        'token_type': tokenType,
        if (userId != null) 'user_id': userId,
      };
}

/// Response model for token refresh (same as AuthenticationResponse).
/// Matches iOS `RefreshTokenResponse` typealias.
typedef RefreshTokenResponse = AuthenticationResponse;
